### DISCUSSION TITLE
Custom dataset and task == custom model?

### DISCUSSION THREAD

**Discussion by asy51** [Project-MONAI/MONAILabel#848] [Q&A]
I am trying to apply the provided deepedit model to DESS MR images to segment knee bones on 3D Slicer using Scribbles, to no avail.

![Screenshot 2022-06-23 131443](https://user-images.githubusercontent.com/90255040/175358323-3ba1079b-1d12-4f05-a74b-ae5cc81f8724.jpg)

When I use SmartEdit/Deepgrow Landmarks, the high intensity muscles (perhaps similar to the spleen in the given model) are selected even though they are far away from where I placed the foreground landmark,

Since the model was trained on spleen masks of CT images, is it not possible to adapt it to my task? Do I need to train a separate model for each specific task? If so, how do I configure my model to interact with MONAILabel?

Thank you for reading.

**Comment by diazandr3s** [comment]
Hi @asy51,

Thanks for opening this discussion. 
The good thing about MONAI Label is that you can easily get a pretrained model. For that, you would need 

1/ Define the label names [here](https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/segmentation.py#L33)
2/ Create labels either by using the scribbles-based algorithm, grow from seeds or any other tools in Slicer. If using scribbles, you may find this conversation useful: https://github.com/Project-MONAI/MONAILabel/issues/447#issuecomment-1159300470
2a/ Submit labels to the server
3/ Trigger the training process for the segmentation, deepgrow or deepedit model. I'd recommend you first start using segmentation model and then with deepedit if needed.

[In this video](https://youtu.be/wtiEe_jiUzg?t=3519) I've shown how you can create labels using grow from seeds and then train a model.

You may not get decent results using the deepgrow model as is because it was trained on abdominal CT images. We did that to demonstrate how the models work.

`Since the model was trained on spleen masks of CT images, is it not possible to adapt it to my task? `

Yes, you should first create some labels

`Do I need to train a separate model for each specific task? If so, how do I configure my model to interact with MONAILabel?`

I'd recommend you follow the steps above. [Here ](https://youtu.be/wtiEe_jiUzg?t=2841)I explained further those steps.

**Reply by sabino-ramirez** [reply]
Hi @diazandr3s 

Great insight. I have been migrating to MONAI Label especially since NVIDIA Clara Train v4.1 and your bootcamp and project week videos have been very helpful. 

For context, I leveraged Clara Train to train a 3d deepgrow model on my own dataset. It performs well and I'd like to start developing further. My main confusion so far has been the relationship between the current AIAA server setup and MONAI Label's. I noticed that in the current version of Clara Train, the workspace is changed to represent the a MONAI Label app. Can I then wrap the model trained on my own data in a MMAR and interact with it using the MONAI Label api and plugins?

Basically, in your opinion, is it even necessary to continue using NVIDIA Clara? I can't help but think I should just "restart" using MONAI Label. Thanks for your feedback!

**Reply by diazandr3s** [reply]
Thanks, @sabino-ramirez.

`For context, I leveraged Clara Train to train a 3d deepgrow model on my own dataset. It performs well and I'd like to start developing further. My main confusion so far has been the relationship between the current AIAA server setup and MONAI Label's. I noticed that in the current version of Clara Train, the workspace is changed to represent the a MONAI Label app. Can I then wrap the model trained on my own data in a MMAR and interact with it using the MONAI Label api and plugins?`

Yes, you can do that. MONAI Label has all the APIs you had in AIAA. Happy to meet and discuss more this, if needed.

`Basically, in your opinion, is it even necessary to continue using NVIDIA Clara? I can't help but think I should just "restart" using MONAI Label. Thanks for your feedback!`

My suggestion is to restart using MONAI Label. One of the great things about MONAI Label is that you can train or update the model. It also has active learning strategies that you can use to increase model performance.

Please don't hesitate to ask. All feedback is welcome :)

**Reply by sabino-ramirez** [reply]
I have since restarted using ML. All has been going well apart from one point that I haven't been able to understand. I start a ML server using the radiology sample app, update the labels object, and set use_pretrained_model to false in lib/configs/deepedit.py in both the init and trainer functions. About half of the datastore is labeled and there is a model trained for 100 epochs on that labeled data. 

In the video, you are able to do inference on the following images using the model trained on your data so far. When I look for the option to use the model to try to annotate the Next Sample image, there is no entry to select it as the target model in any of the method tabs. However, I'm able to use the model for inference on an image using the api endpoint. Slicer logs also show 4 models available including 1 Deepedit and 1 Segmentation.

Do you have an idea of what I could be doing wrong? Or maybe I'm wrong about the expected functionality. 
Either way thanks for reading!

**Reply by diazandr3s** [reply]
Thanks, @sabino-ramirez 

> 
> When I look for the option to use the model to try to annotate the Next Sample image, there is no entry to select it as the target model in any of the method tabs. However, I'm able to use the model for inference on an image using the api endpoint. Slicer logs also show 4 models available including 1 Deepedit and 1 Segmentation.
> 
> Do you have an idea of what I could be doing wrong? Or maybe I'm wrong about the expected functionality.

This is a known issue that should be solved by downloading the latest preview Slicer. Please check this https://github.com/Project-MONAI/MONAILabel/issues/852 and this https://github.com/Project-MONAI/MONAILabel/issues/852#issuecomment-1172436198

Hope that helps
